### ISSUE TITLE
hostd error on container that we created: error on file delete force

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#5648]
This is coming off of a container created here:
docker %{VCH-PARAMS} create ${busybox} ls

This is from 1-10-Docker-PS test.

The test does not have any docker cleanup commands in it, so this must be coming from the vic-machine delete at the end of the test:

```
2017-07-06T13:02:57.995Z info hostd[34080B70] [Originator@6876 sub=Vimsvc.TaskManager opID=c3cfea92 user=droneci] Task Created : haTask--vim.VirtualDiskManager.deleteVirtualDisk-26069
2017-07-06T13:02:57.996Z error hostd[332C1B70] [Originator@6876 sub=Nfcsvc opID=c3cfea92 user=droneci] file delete force (/vmfs/volumes/58af7eb5-d9fc5174-6127-0cc47a9be492/5dfa31f53d754f0996522c2c0e0a051c1ce2dc0d2924586d911e4dc998d2a228/5dfa31f53d754f0996522c2c0e0a051c1ce2dc0d2924586d911e4dc998d2a228-delta.vmdk -> /vmfs/volumes/58af7eb5-d9fc5174-6127-0cc47a9be492/5dfa31f53d754f0996522c2c0e0a051c1ce2dc0d2924586d911e4dc998d2a228/5dfa31f53d754f0996522c2c0e0a051c1ce2dc0d2924586d911e4dc998d2a228-delta.vmdk) by 'droneci'
2017-07-06T13:02:57.996Z info hostd[332C1B70] [Originator@6876 sub=Vimsvc.TaskManager opID=c3cfea92 user=droneci] Task Completed : haTask--vim.VirtualDiskManager.deleteVirtualDisk-26069 Status error
```

**Comment by mhagen-vmware** [comment]
[vmsupport-bundle.zip](https://github.com/vmware/vic/files/1128366/vmsupport-bundle.zip)

**Comment by corrieb** [comment]
Mike should this be in "New Issues"? Don't want to question the master but...

**Comment by dougm** [comment]
Looks like this message is caused by:
- Disks are detached from the VM before destroying
- Datastore files are removed 1-by-1
- Any .vmdk file is deleted via VirtualDiskManager, which deleted both the descriptor (foo.vmdk) and backing file (foo-delta.vmdk)
- The error happens when trying to remove the backing file directly after it was already removed

We are deleting files 1-by-1 to workaround an old vsan bug (directory can't be removed with .vmdk deletable=false).  We could refactor the code to limit the search spec to VmDiskFileQuery, which would just return the .vmdk descriptors.  Delete those files, then delete the directory to remove the rest.

In any case, the error is benign.

**Comment by dougm** [comment]
Implemented vmware/govmomi#810 so we can the proposed change properly, but the vic-machine changes were more involved than expected.  Opting for a smaller change in the short term in PR #6040
